### PR TITLE
networkd/wireguard: support network.wireguard.* credentials

### PR DESCRIPTION
**Pull request by YHNdnzj**
Closes https://github.com/systemd/systemd/issues/26702

### CODE REVIEW

**Comment by poettering** [comment]
I think this is good as as it is, but I wonder if we should add one more thing on top: if now keys are configured explicitly, and we need one, automatically load it from `network.wireguard.public.<basenameofconfigfile>` or so?

i.e. that in `50-foobar.network`, if you don't configure any wireguard keys, we'd automatically look for `network.wireguard.public.50-foobar` and so on.

i.e. not just allow people to import they keys but also suggests a uniform naming scheme to them.

**Comment by YHNdnzj** [comment]
Updated. PTAL.

File: src/network/netdev/wireguard.c [hunk 1]
```diff
@@ -1088,10 +1125,34 @@ static int wireguard_verify(NetDev *netdev, const char *filename) {
...
+                                                      "%s: No private key specified and default key isn't available, "
+                                                      "ignoring network device: %m",
+                                                      filename);
+
+                if (cred_len != WG_KEY_LEN)
+                        return log_netdev_error_errno(netdev, r,
+                                                      "%s: No private key specified and default key is invalid. Ignoring network device.",
+                                                      filename);
+
+                memcpy(w->private_key, cred, WG_KEY_LEN);
```
Comment (yuwata) [inline src/network/netdev/wireguard.c]: Is it not necessary to decode the cred?

Comment (YHNdnzj) [reply src/network/netdev/wireguard.c]: Oops...

**Review by yuwata — APPROVED** [review]

File: units/systemd-networkd.service.in [hunk 2]
```diff
@@ -50,6 +50,7 @@ SystemCallErrorNumber=EPERM
 SystemCallFilter=@system-service
 Type=notify-reload
 User=systemd-network
+ImportCredential=network.wireguard.*
```
Comment (poettering) [inline units/systemd-networkd.service.in:53]: can you please also briefly list this in systemd.system-credentials(7)

**Comment by YHNdnzj** [comment]
CI finally passed. Yay!

**Pull request merged**